src/main/java/io/fabric8/maven/docker/assembly/DockerFileBuilder.java

```diff
@@ -235,14 +237,48 @@ private String quote(String value) {
 
     private void addPorts(StringBuilder b) {
         if (ports.size() > 0) {
-            String[] portsS = new String[ports.size()];
-            int i = 0;
-            for (Integer port : ports) {
-                portsS[i++] = port.toString();
+            String[] portsS = ports.toArray(new String[]{});
+            for(String port : portsS) {
+            	validatePortMapping(port);
             }
             DockerFileKeyword.EXPOSE.addTo(b, portsS);
         }
     }
+    
+    // Pattern for splitting of the protocol part of the port.  
+    private static final Pattern PROTOCOL_SPLIT_PATTERN = Pattern.compile("(.*?)(?:/(tcp|udp))?$");
+    private void validatePortMapping(String input) throws IllegalArgumentException {
+        try {
+            Matcher matcher = PROTOCOL_SPLIT_PATTERN.matcher(input);
+            // Matches always.  If there is a tcp/udp protocol, should end up in the second group
+            // and get factored out.  If it's something invalid, it should get stuck to the first group.  
+            matcher.matches();
+            
+            // First group is the port mapping.  
+            // It may be split with a : in a variety of ways.  
+            String mapping = matcher.group(1);
+            String[] mappingParts = mapping.split(":", 3);
```

Comment:
Just re-looked at this PR and I think that is not correct here as we are talking about the ports to expose in a Dockerfile and not about mappings. So there must not be a `:` inlcude in the specificiation and only `8080` and `8080/tcp` should be allowed.